apps/aehttp/src/aehttp_dispatch_ext.erl

```diff
@@ -28,14 +28,45 @@ handle_request('Ping', #{source := Source,
 
 handle_request('GetTop', _, _Context) ->
     {ok, Header} = aec_chain:top_header(),
-    {ok, Resp} = aec_headers:serialize_to_map(Header),
-    {200, [], Resp};
+    {ok, HH} = aec_headers:hash_header(Header),
+    {ok, Resp} = aec_headers:serialize_to_map(
+                   aec_headers:replace_empty_pow(Header)),
+    {200, [], maps:put(hash, base64:encode(HH), Resp)};
 
-handle_request('GetBlock', Req, _Context) ->
-    Hash = maps:get('BlockHash', Req),
-    {ok, Block} = aec_chain:get_block_by_hash(aeu_hex:hex_to_bin(Hash)),
-    {ok, Resp} = aec_blocks:serialize_for_network(Block),
-    {200, [], Resp};
+handle_request('GetBlockByHeight', Req, _Context) ->
+    Height = maps:get('height', Req),
+    case aec_chain:get_block_by_height(Height) of
+        {ok, Block} ->
+            %% swagger generated code expects the Resp to be proplist or map
+            %% and always runs jsx:encode/1 on it - even if it is already
+            %% encoded to a binary; that's why we use
+            %% aec_blocks:serialize_to_map/1 instead of
+            %% aec_blocks:serialize_for_network/1 
+            Resp = aec_blocks:serialize_to_map(
+                     aec_blocks:replace_empty_pow(Block)),
+            {200, [], Resp};
+        {error, {chain_too_short, _}} ->
+            {404, [], #{reason => <<"chain too short">>}}
+    end;
+
+handle_request('GetBlockByHash', Req, _Context) ->
+    Hash = base64:decode(maps:get('hash', Req)),
+    {ok, Header} = aec_chain:get_header_by_hash(Hash),
```

Comment:
Of course this can fail too. Specifying a non-existing hash would cause the client to see e.g. 500.